### PR TITLE
Format small

### PR DESCRIPTION
**Pull request by uribgp**
Related to #1496

Description: Breaking up large pull

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by uribgp** [comment]
It is returning this error: 
`
error: Unsupported operand types for * ("int" and "None")  [operator]
`

for this code in /ignite/ignite/engine/engine.py:
`
warnings.warn(
                                "Data iterator can not provide data anymore but required total number of "
                                "iterations to run is not reached. "
                                f"Current iteration: {self.state.iteration} vs Total iterations to run : {self.state.epoch_length * self.state.max_epochs}" # type: ignore[operator]
                            )
`

Here is the original:
`
                            warnings.warn(
                                "Data iterator can not provide data anymore but required total number of "
                                "iterations to run is not reached. "
                                "Current iteration: {} vs Total iterations to run : {}".format(
                                    self.state.iteration,
                                    self.state.epoch_length * self.state.max_epochs,  # type: ignore[operator]
                                )
                            )
`

**Comment by vfdev-5** [comment]
It's mypy complains about how you split the string. Annotation `# type: ignore[operator]` should be on the last string:
```python
                            warnings.warn(
                                "Data iterator can not provide data anymore but required total number of "
                                "iterations to run is not reached. "
                                f"Current iteration: {self.state.iteration} vs Total iterations to run : {self.state.epoch_length * self.state.max_epochs}"  # type: ignore[operator]
                            )
```

**Comment by uribgp** [comment]
> It's mypy complains about how you split the string. Annotation `# type: ignore[operator]` should be on the last string:


Thanks.  I have that locally, and my commit on github shows it that way.  I believe it is the "autopep8 fix" which is formatting it that way for some reason.

**Comment by vfdev-5** [comment]
> Thanks. I have that locally, and my commit on github shows it that way. I believe it is the "autopep8 fix" which is formatting it that way for some reason.

Yeah, I saw that... Can you find out who does that: autopep8 or black ?
I filed an issue to fix `# type: ignore[operator]` if possible. Maybe, we can simply create a substring with `# type: ignore[operator]` and finally concat it to the complete warning message...

**Comment by uribgp** [comment]
It is autopep8 

If I concat it, wouldn't it show up as a string?

**Comment by vfdev-5** [comment]
Here is what I was thinking about : 
```python
msg = f"Current iteration: {self.state.iteration} vs Total iterations to run : {self.state.epoch_length * self.state.max_epochs}"  # type: ignore[operator]

warnings.warn(
    "Data iterator can not provide data anymore but required total number of "
    f"iterations to run is not reached. {msg}"
)
```

**Comment by uribgp** [comment]
Very clever, I'll give it a shot! #

**Comment by vfdev-5** [comment]
autopep8 is unmercy :(
Let's try then the following:
```python
total_iters = self.state.epoch_length * self.state.max_epochs  # type: ignore[operator]
msg = f"Current iteration: {self.state.iteration} vs Total iterations to run : {total_iters}"

warnings.warn(
    "Data iterator can not provide data anymore but required total number of "
    f"iterations to run is not reached. {msg}"
)
```

**Comment by uribgp** [comment]
Yeah it's brutal!  Giving it a shot.

**Comment by vfdev-5** [comment]
@uribgp please merge your branch with master. @gruebel fixed this in #1519.

**Comment by uribgp** [comment]
OK, done.

File: ignite/metrics/epoch_metric.py [hunk 1]
```diff
@@ -88,15 +88,12 @@ def _check_type(self, output: Tuple[torch.Tensor, torch.Tensor]) -> None:
         dtype_preds = self._predictions[-1].dtype
         if dtype_preds != y_pred.dtype:
             raise ValueError(
-                "Incoherent types between input y_pred and stored predictions: "
-                "{} vs {}".format(dtype_preds, y_pred.dtype)
+                "Incoherent types between input y_pred and stored predictions: " f"{dtype_preds} vs {y_pred.dtype}"
```
Comment (vfdev-5) [inline ignite/metrics/epoch_metric.py:91]:
```suggestion
                f"Incoherent types between input y_pred and stored predictions: {dtype_preds} vs {y_pred.dtype}"
```

**Review by vfdev-5 — APPROVED** [review]
Looks good, thanks @uribgp !

**Pull request merged**